.github/workflows/dev_builds.yml

```diff
@@ -27,7 +27,14 @@ jobs:
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
-      - name: Push kubernetes-tools image
+      - name: Push kubernetes-tools image to Dockerhub
         run: make push-image BUILD_TAG=${{ steps.extract_tag.outputs.tag }}
-      - name: Push kubernetes-tools image build cache
+      - name: Push kubernetes-tools image build cache to Dockerhub
         run: make push-image-cache BUILD_TAG=${{ steps.extract_tag.outputs.tag }}
+      - name: Login to ECR
+        run: make login-ecr
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      - name: Build and push image to ECR
+        run: make push-image-ecr BUILD_TAG=${{ steps.extract_tag.outputs.tag }}
```

Comment:
This workflow pushes the build cache images to Docker Hub, but you are not pushing the build cache images to ECR in this workflow, even though you do it in the `release` workflow. Is there a specific reason for this difference?

As a side note (not really related to this pull request), I don't understand why the `dev` and `release` workflows build the image _not_ using the cache (by running `make build-release-image`) _and_ they both publish the build cache (to Docker Hub), but the `pre-release` workflow _does_ use cache while building, but does _not_ publish the built cache. Perhaps @pmalek-sumo can help here, giving more context on this change https://github.com/SumoLogic/sumologic-kubernetes-tools/commit/26bf77b7f4fa5555bbd73940a79f56f50f043f8f?)

Reply:
I added the missing cache push to ECR, so we should be able to merge this with consistent behaviour between Docker Hub and ECR. If we want to make improvements to the CI logic and structure, I think we should do it in a separate change.

But yeah, it's rather messy. To begin with, the only difference between `build-image` and `build-release-image` is the cache, which we should be able to always use safely. We should also only push to the cache in one place IMO, probably in the dev workflow, which is fired more frequently.

Incidentally, we have three workflows which are nearly identical, and since I already messed up making synchronous changes to them, I think we should make them DRY via a reusable workflow with some flags. That would also make explicit where we're pushing the cache and tagging latest.

Reply:
💯  Agree on all your points @swiatekm-sumo

Reply:
@astencel-sumo see https://github.com/SumoLogic/sumologic-kubernetes-tools/pull/166